ocs_ci/ocs/node.py

```diff
@@ -1027,3 +1027,18 @@ def node_replacement_verification_steps_ceph_side(old_node_name, new_node_name):
 
     log.info("Verification steps from the ceph side finish successfully")
     return True
+
+
+def get_worker_hostname():
+    """
+    Gets the worker node hostnames
+
+    Returns:
+        list: List of worker node hostnames
+
+    """
+    worker_node_objs = get_typed_nodes()
+    return [
+        worker_obj.get()['metadata']['labels'][constants.HOSTNAME_LABEL]
+        for worker_obj in worker_node_objs
+    ]
```

Comment:
Could you fix this in the same function "get_compute_node_names" or remove one function.

By looking into names "get_worker_hostname" and "get_compute_node_names" , both looks like similar behavior. We need to maintain single function if the behavior is same.

Reply:
if we go with get_compute_node_names for baremetal,it will replace hostname with "-" for eg:- hostname of node "bm.woker.one.test.openshift" return value from above function is "bm-worker-one-test-openshift" this return value does not match the node selector for host and that will cause ocs install failure

Reply:
can we add a condition to `get_compute_node_names` to return in the desired format in case needed?

Reply:
if we add there it will get more complicated because the same function is used for verification of OSD's and hostname

Reply:
reason for creating a new function was the return value was we get from those function 

Return Value from get_compute_node_names
> INFO - ['argo005-ceph-redhat-com', 'argo006-ceph-redhat-com', 'argo007-ceph-redhat-com']

Return Value from get_worker_hostname
> INFO - ['argo005.ceph.redhat.com', 'argo006.ceph.redhat.com', 'argo007.ceph.redhat.com']

Expected return value for using as selector
```
$ oc get nodes -o json |jq -r '.items|.[].metadata.labels."kubernetes.io/hostname"'
argo002.ceph.redhat.com
argo003.ceph.redhat.com
argo004.ceph.redhat.com
argo005.ceph.redhat.com
argo006.ceph.redhat.com
argo007.ceph.redhat.com
```
get_worker_hostname will return the expected value that's the reason created the new function

Reply:
Switch to old function added extra param not to replace hostname